File: games/inflation-rpg/native/onestore-iap/ios/Plugin/OnestoreIapPlugin.m
#import <Foundation/Foundation.h>
#import <Capacitor/Capacitor.h>

CAP_PLUGIN(OnestoreIapPlugin, "OnestoreIap",
           CAP_PLUGIN_METHOD(initialize, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(queryProducts, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(purchase, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(acknowledge, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(restorePurchases, CAPPluginReturnPromise);
)

File: games/inflation-rpg/native/onestore-iap/ios/Plugin/OnestoreIapPlugin.swift
import Foundation
import Capacitor

@objc(OnestoreIapPlugin)
public class OnestoreIapPlugin: CAPPlugin {
    @objc func initialize(_ call: CAPPluginCall) {
        call.reject("ONE store IAP is Android only. Use StoreKit on iOS — wired in a later phase.")
    }

    @objc func queryProducts(_ call: CAPPluginCall) {
        call.reject("ONE store IAP is Android only")
    }

    @objc func purchase(_ call: CAPPluginCall) {
        call.reject("ONE store IAP is Android only")
    }

    @objc func acknowledge(_ call: CAPPluginCall) {
        call.reject("ONE store IAP is Android only")
    }

    @objc func restorePurchases(_ call: CAPPluginCall) {
        call.reject("ONE store IAP is Android only")
    }
}
